src/api/route-services/work-response.js

```diff
@@ -106,15 +103,16 @@ class WorkResponseService {
       if (pagination) {
         responseForClient.results = handlePagination(processedResults, pagination);
       }
-
-      if (Date.parse(timeout) > Date.now()) {
-        this.io.to(socketId).emit(`WorkResponse-${uuid}`, responseForClient);
-      }
-
-      logger.log('response sent out');
     } catch (e) {
       logger.error('Error trying to cache or paginate data: ', e);
+      return;
     }
+
+    if (Date.parse(timeout) > Date.now()) {
+      this.io.to(socketId).emit(`WorkResponse-${uuid}`, responseForClient);
+    }
+
+    logger.log('Work response sent out.');
```

Comment:
even if the timeout has passed, the logs will still say `Work response sent out.` which will be wrong. To fix this, better put the logger.log line inside the if statement and have an else statement that is a log message saying that timeout has expired?

Reply:
good point, fixed